src/index.js

```diff
@@ -1,26 +1,73 @@
 import './style.css';
 
 const list = document.querySelector('.list');
+const enter = document.querySelector('.enter');
+const input = document.querySelector('.task_input');
+const array = JSON.parse(localStorage.getItem('array')) || [];
 
-const array = [
-  {
-    description: 'clean house',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'feed animals',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'go shopping',
-    completed: false,
-    index: 2,
-  },
-];
-
-array.forEach((task) => {
-  const item = `<li class="item"><div><input type="checkbox">${task.description}</div><a class="move" href="#"><img alt= "move" src="../../../icons/vector3.png"></a></li>`;
-  list.insertAdjacentHTML('beforeend', item);
-});
\ No newline at end of file
+function update(index, value) {
+  if (value !== '') {
+    array[index] = { description: value, completed: false };
+    localStorage.setItem('array', JSON.stringify(array));
+  }
+}
+
+function removeTask(index) {
+  array.splice(index, 1);
+  localStorage.setItem('array', JSON.stringify(array));
+}
+
+function display() {
+  list.innerHTML = '';
+  array.forEach((task) => {
+    const item = `<li class="item"><div class ="check-cont"><input class="checkbox" type="checkbox">
+    <input class="edit" id="${array.indexOf(task)}" type="text" value = "${task.description}"></div>
+    <a class="delete hide" href="#"><img alt= "delete" src="../../../icons/bin.png"></a>
+    <a class="move" href="#"><img alt= "move" src="../../../icons/vector3.png"></a></li>`;
+    list.insertAdjacentHTML('beforeend', item);
+  });
+  document.querySelectorAll('.edit').forEach((task) => {
+    const remove = task.parentElement.nextElementSibling;
+    task.onfocus = function () {
+      remove.classList.remove('hide');
+    };
+
+    remove.addEventListener('click', (e) => {
+      const index = e.target.parentElement.previousElementSibling.firstChild.nextElementSibling.getAttribute(('id'), 10);
+      removeTask(index);
+      display();
+    });
+
+    task.addEventListener('keyup', (e) => {
+      if (e.key === 'Enter') {
+        const index = parseInt(e.target.getAttribute('id'), 10);
+        update(index, task.value);
+        display();
+      }
+    });
+  });
+}
+
+function add() {
+  if (input.value !== '') {
+    array.push({
+      description: input.value,
+      completed: false,
+    });
```

Comment:
- [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md) for this project, " `By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).` ", your tasks(`todos`) are missing the index key(property). Please kindly add it, and it should be set to the value of the new array length.
- [ ] Please also ensure deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.